cartoframes/io/managers/context_manager.py

```diff
@@ -294,13 +312,22 @@ def _truncate_table(self, table_name, schema, cartodbfy):
 
     def _truncate_and_drop_add_columns(self, table_name, schema, df_columns, table_columns, cartodbfy):
         log.debug('TRUNCATE AND DROP + ADD columns table "{}"'.format(table_name))
-        query = '{regenerate}; BEGIN; {truncate}; {drop_columns}; {add_columns}; {cartodbfy}; COMMIT;'.format(
+        drop_columns = _drop_columns_query(table_name, table_columns)
+        add_columns = _add_columns_query(table_name, df_columns)
+
+        drop_add_columns = 'ALTER TABLE {table_name} {drop_columns},{add_columns};'.format(
+            table_name=table_name, drop_columns=drop_columns, add_columns=add_columns)
+
+        qualified_func_name = self._create_function(schema=schema, statement=drop_add_columns)
+        drop_add_func_sql = 'SELECT {qualified_func_name}'.format(qualified_func_name=qualified_func_name)
+
+        query = '{regenerate}; BEGIN; {truncate}; {drop_add_func_sql}; {cartodbfy}; COMMIT;'.format(
             regenerate=_regenerate_table_query(table_name, schema) if self._check_regenerate_table_exists() else '',
             truncate=_truncate_table_query(table_name),
-            drop_columns=_drop_columns_query(table_name, table_columns),
-            add_columns=_add_columns_query(table_name, df_columns),
+            drop_add_func_sql=drop_add_func_sql,
             cartodbfy=_cartodbfy_query(table_name, schema) if cartodbfy else '')
         self.execute_long_running_query(query)
+        self._delete_function(qualified_func_name)
```

Comment:
Why is the delete_function done here and not in end of the same transaction?

Reply:
Initially thought it'd be a good idea to leave the UDF creation/deletion aside from the transaction for 'consistency' as creation would always need to be made beforehand through a normal SQL API call, but after reviewing it again I've changed it so that it is done within the same transaction as commented (avoiding one extra API call).

I've left the `_delete_function` in case it'd be useful for any future usage but please let me know if you'd rather remove it completely as it's not being currently used anymore.

Reply:
What you prefer.

This is a tricky point because if the batch request fails the tmp function won't be removed, so maybe we can catch the error and run _delete_function then.

Reply:
That'd be right, maybe we could leave it outside from the transaction and perform something similar to the following to ensure that the temp UDF gets deleted in any case, what do you think? 

```python
try:
    self.execute_long_running_query(query)
finally:
    self._delete_function(qualified_func_name)
```

Reply:
Yeah. Makes sense